### PR TITLE
Add circleci config with slack notification

### PR DESCRIPTION
**Pull request by ampersarnie**
## Description
To help keep everyone within Big Bite informed of any releases that are happening with PHPCS configs, it might be a good idea to use Slack webhooks to notify specific channels of any releases. This has already been added to the [Build Tools repo](https://github.com/bigbite/build-tools).

CircleCI will need enabling on the repo and `SLACK_WEBHOOK` needs to be set in the environment variables - I can provide the webhook url, just reach out.

## Changelog
* Adds a CircleCI config to the project to allow any releases to notify Big Bite slack channels.

## Types of changes (_if applicable_):

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist (_if applicable_):
- [x] All checks pass when running `composer run all-checks.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  node: circleci/node@4.7.0
+
+jobs:
+  notify_slack:
+    docker:
+      - image: cimg/node:17.2.0
+    steps:
+      - run:
+          command: |
+            curl -X POST -H 'Content-type: application/json' --data '{"text":"New PHPCS Config Release","blocks":[{"type":"header","text":{"type":"plain_text","text":"New Build Tools Release"}},{"type":"section","fields":[{"type":"mrkdwn","text":"*Version:* <https://github.com/'${CIRCLE_PROJECT_USERNAME}'/'${CIRCLE_PROJECT_REPONAME}'/releases/tag/'${CIRCLE_TAG}'|'${CIRCLE_TAG}'>"},{"type":"mrkdwn","text":"<https://github.com/'${CIRCLE_PROJECT_USERNAME}'/'${CIRCLE_PROJECT_REPONAME}'/issues|Issues> *|* <https://github.com/'${CIRCLE_PROJECT_USERNAME}'/'${CIRCLE_PROJECT_REPONAME}'/pulls|Pull Requests>"}]}]}' ${SLACK_WEBHOOK}
```
Comment (jonmcp) [inline .circleci/config.yml:13]: Text is incorrect - it mentions the build tools, not this package

**Review by jonmcp — DISMISSED** [review]
Thanks for the PR, this is a nice idea. Just one thing that will need addressing before we merge.
Can you sort out the webhook URL for me as well, please, as I'm unfamiliar with Circle CI?

**Comment by ampersarnie** [comment]
> Thanks for the PR, this is a nice idea. Just one thing that will need addressing before we merge. Can you sort out the webhook URL for me as well, please, as I'm unfamiliar with Circle CI?

Ta, forgot it was in there twice. Yeah no problem I can sort out the webhook and so on.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  node: circleci/node@4.7.0
+
+jobs:
+  notify_slack:
+    docker:
+      - image: cimg/node:17.2.0
+    steps:
+      - run:
+          command: |
+            curl -X POST -H 'Content-type: application/json' --data '{"text":"New PHPCS Config Release","blocks":[{"type":"header","text":{"type":"plain_text","text":"New  PHPCS Config Release"}},{"type":"section","fields":[{"type":"mrkdwn","text":"*Version:* <https://github.com/'${CIRCLE_PROJECT_USERNAME}'/'${CIRCLE_PROJECT_REPONAME}'/releases/tag/'${CIRCLE_TAG}'|'${CIRCLE_TAG}'>"},{"type":"mrkdwn","text":"<https://github.com/'${CIRCLE_PROJECT_USERNAME}'/'${CIRCLE_PROJECT_REPONAME}'/issues|Issues> *|* <https://github.com/'${CIRCLE_PROJECT_USERNAME}'/'${CIRCLE_PROJECT_REPONAME}'/pulls|Pull Requests>"}]}]}' ${SLACK_WEBHOOK}
```
Comment (jonmcp) [inline .circleci/config.yml:13]: Sorry to be a pest, but there's now a double space in the message text 🙈 Might look a bit weird in Slack.

**Review by jonmcp — DISMISSED** [review]
Thanks for the fast turnaround on the feedback. Sorry, there's a little more

File: .circleci/config.yml (hunk 2 above)
Comment (ampersarnie) [reply .circleci/config.yml]: This what I get for rushing.

**Review by jonmcp — APPROVED** [review]
thanks!

**Pull request merged**